### PR TITLE
A change which returns focus back to terminal when switching branches from the terminal

### LINKED ISSUE #6: when switching branches, focus moves from integrated terminal
when switching branches using the integrated vs code terminal, the focus moves away from the terminal, meaning you have to click on it in in order to run commands again.

it's very common that the first thing one does after switching branches is a pull, merge etc, so having to click into the terminal slows down workflow a little. retaining focus on the terminal is also default vs code behaviour when switching branches without this extension.

### PR DESCRIPTION
**Pull request by romunov**
A while ago, a user in [issues #6 reported](https://github.com/avaly/vscode-restore-git-branch-tabs/issues/6) that when using the terminal to switch branches, the focus is taken away from the terminal once all the editor tabs are loaded. Until recently I used Ctrl+J+J to minimize and restore the terminal again, but it seemed kind of clunky. This PR would solve and close #6.

In this pull request, I implemented a way for the plugin to move the focus (back) to the terminal on branch switch. Unfortunately I couldn't find a way to determine if the switching was done through the terminal or not (the API returns terminal that has been used (recently)) so that I could handle the return of the focus more reliably, but in my testing, this is behaving well enough. If a user is using the GUI to switching branches, they're mouse-clicking tabs they want to edit anyway.

Please give this PR a test and let me know if there's anything you'd like to modify. Since I'm new to this TypeScript business I'm not exactly sure what the etiquette is regarding `tsconfig.json`, `package-lock.json` and `package.json`. Any guidance on the matter would be much appreciated.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,10 +117,10 @@
     "@types/node": "^10.12.21",
     "@types/vscode": "^1.38.0",
     "glob": "^7.1.4",
-    "mocha": "^6.1.4",
-    "typescript": "^4.6.3",
+    "mocha": "^10.2.0",
     "tslint": "^5.12.1",
-    "vsce": "2.7.0",
+    "typescript": "^4.6.3",
+    "vsce": "^1.97.0",
```
Comment (avaly) [inline package.json:123]: Please do not downgrade packages without a specific reason. `vsce` should be upgraded only.

File: src/savedEditor.ts [hunk 2]
```diff
@@ -28,9 +30,26 @@ export class SavedEditor {
...
 
+        let cfg = workspace.getConfiguration().get<IConfig>(ExtensionKey);
+        if (cfg === undefined) {
+            cfg = defaultIConfig;
+        }
+        let cfgTerminalFocus = cfg.returnFocusOnTerminal
+
         Logger.log(`SavedEditor.open: Opening this <${this}>`);
 
+        const active_terminal = window.activeTerminal
```
Comment (avaly) [inline src/savedEditor.ts:41]: Please use existing naming patterns:

```suggestion
        const { activeTerminal } = window;
```

And also update the lines below with this variable name.

File: package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "restore-git-branch-tabs-improved",
   "displayName": "Restore Git Branch Tabs Improved",
   "description": "Remembers and restores which tabs were opened for each git branch.",
-  "version": "1.1.0",
+  "version": "1.2.0",
```
Comment (avaly) [inline package.json:5]: Please revert this change as well. I'll bump the version after merging this PR.

File: src/savedEditor.ts [hunk 4]
```diff
@@ -28,9 +30,26 @@ export class SavedEditor {
             preview: false
         };
 
+        let cfg = workspace.getConfiguration().get<IConfig>(ExtensionKey);
+        if (cfg === undefined) {
+            cfg = defaultIConfig;
+        }
+        let cfgTerminalFocus = cfg.returnFocusOnTerminal
```
Comment (avaly) [inline src/savedEditor.ts:37]: Re-use the variable name via destructuring:

```suggestion
        let { returnFocusOnTerminal } = cfg;
```

**Review by avaly — CHANGES_REQUESTED** [review]

**Comment by avaly** [comment]
@romunov I fixed this in a different way in f0880ee.

**Comment by romunov** [comment]
Excellent!

**Pull request closed without merging**